syntax = "proto3";

// =====================================================================
// GET ALL Payment Service
// =====================================================================

message PaymentRequest {
    int32 user_id = 1;
}

message Payment {
    int32 id = 1;
    int32 user_id = 2;
    int32 plan_id = 3;
    int32 method_id = 4;
    float total = 5;
}

message PaymentResponse {
    repeated Payment payment = 1;
}

service PaymentService {
    rpc GetPaymentData (PaymentRequest) returns (PaymentResponse);
}


// =====================================================================
// GET ALL Plan Service
// =====================================================================

message PlanRequest {
    int32 plan_id = 1;
}

message PlanResponse {
    int32 id = 1;
    string name = 2;
    int32 value = 3;
}

service PlanService {
    rpc GetPlanData (PlanRequest) returns (PlanResponse);
}

// =====================================================================
// GET ALL Method Service
// =====================================================================

message MethodRequest {
    int32 method_id = 1;
}

message MethodResponse {
    int32 id = 1;
    string name = 2;
    float discount = 3; 
}

service MethodService { 
    rpc GetMethodData (MethodRequest) returns (MethodResponse);
}